include/swift/AST/Attr.def

```diff
@@ -347,21 +361,23 @@ SIMPLE_DECL_ATTR(testable, Testable,
 DECL_ATTR(_alignment, Alignment,
   OnStruct | OnEnum |
   UserInaccessible |
-  ABIStableToAdd | ABIStableToRemove | APIStableToAdd | APIStableToRemove,
+  ABIBreakingToAdd | ABIBreakingToRemove | APIStableToAdd | APIStableToRemove,
   56)
 SIMPLE_DECL_ATTR(rethrows, Rethrows,
   OnFunc | OnAccessor | OnConstructor |
   RejectByParser |
-  ABIStableToAdd | ABIStableToRemove | APIStableToAdd | APIStableToRemove,
+  ABIStableToAdd | ABIBreakingToRemove | APIStableToAdd | APIBreakingToRemove,
   57)
 DECL_ATTR(_swift_native_objc_runtime_base, SwiftNativeObjCRuntimeBase,
   OnClass |
   UserInaccessible |
-  ABIStableToAdd | ABIStableToRemove | APIStableToAdd | APIStableToRemove,
+  ABIBreakingToAdd | ABIBreakingToRemove | APIStableToAdd | APIStableToRemove,
   59)
 CONTEXTUAL_SIMPLE_DECL_ATTR(indirect, Indirect, DeclModifier |
   OnEnum | OnEnumElement |
-  ABIStableToAdd | ABIStableToRemove | APIStableToAdd | APIStableToRemove,
+  // FIXME: Sometimes API-breaking to remove too, if a cycle of enum cases can
+  // be formed with generics.
```

Comment:
I don't think so - if 'indirect' is required it's because the type's cases directly reference themselves.

If the type is generic, one of its cases could store another generic instantiation of itself, but I don't think it can store the same _exact_ instantiation because that would mean the actual substituted type of the case is infinite - `Foo<Foo<Foo<...`. Is that not right?

Reply:
I was thinking you could have a cycle between two types:

```swift
// Library
indirect enum MyOptional<T> {
  case some(T)
  case none
}
```

```swift
enum Evil {
  case lawful
  case chaotic(MyOptional<Evil>)
}
```

But maybe the generic already forces indirection?